node/src/actors/session/handlers.rs

```diff
@@ -179,14 +179,48 @@ impl StreamHandler<Result<BytesMut, Error>> for Session {
 
                                 try_consolidate_session(self, ctx);
                             }
-                            Err(err) => {
+                            Err(err @ HandshakeError::DifferentEpoch{..})
+                            | Err(err @ HandshakeError::DifferentTimestamp{..}) => {
+                                let peers_manager_addr = PeersManager::from_registry();
+                                // Ice the peer that was an error
+                                peers_manager_addr.do_send(RemoveAddressesFromTried {
+                                    addresses: vec![self.remote_addr],
+                                    ice: true,
+                                });
+
                                 if session_type == SessionType::Feeler {
+                                    log::debug!(
+                                        "Dropping feeler connection {}: {}",
+                                        self.remote_addr,
+                                        err
+                                    );
+                                } else {
+                                    log::warn!(
+                                        "Dropping {:?} peer {}: {}",
+                                        session_type,
+                                        self.remote_addr,
+                                        err
+                                    );
+                                }
+
+                                // Stop this session
+                                ctx.stop();
+                            }
+                            Err(err) => {
+                                if session_type == SessionType::Feeler
+                                    || session_type == SessionType::Outbound
+                                {
                                     let peers_manager_addr = PeersManager::from_registry();
-                                    // Ice the peer that was an error
-                                    peers_manager_addr.do_send(RemoveAddressesFromTried {
-                                        addresses: vec![self.remote_addr],
-                                        ice: true,
-                                    });
+                                    // Ice the peer that was an error (except epochs where superblock is updated)
+                                    if self.current_epoch % 10 != 0 {
```

Comment:
I would appreciate some color on why this is needed / what exactly we want to prevent by introducing this check,